src/lms-ds-loader/lms_ds_loader/file_processor.py

```diff
@@ -0,0 +1,40 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+from lms_ds_loader.csv_to_sql import CsvToSql
+from lms_ds_loader.constants import Constants
+
+
+class FileProcessor:
+    """
+    Processes all files in the LMS-UDM compatible file system and calls a
+    program to read and load them into a database.
+
+    Parameters
+    ----------
+    file_system : LmsFilesystemProvider
+        A fully-prepared instance of `LmsFilesystemProvider`.

```

Comment:
I thought about combining `FileProcessor` and `LmsFilesystemProvider` into one class. Single Responsibility Principle encourages me to keep two classes. 

As we look at building analyzers, I'm thinking of moving `LmsFilesystemProvider` to a stand-alone package so that it can be used by those analyzers. `FileProcessor` might become a strategy class that is injected into `LmsFileystemProvider`, or it might stay the way it is right now. No reason to decide that right now.